primitives/runtime-interface/test/src/lib.rs

```diff
@@ -104,30 +110,52 @@ fn host_function_not_found() {
 		\\\"Invalid utf8 data provided\\\")) }\""
 )]
 fn test_invalid_utf8_data_should_return_an_error() {
-	call_wasm_method::<HostFunctions>("test_invalid_utf8_data_should_return_an_error");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_invalid_utf8_data_should_return_an_error");
 }
 
 #[test]
 fn test_overwrite_native_function_implementation() {
-	call_wasm_method::<HostFunctions>("test_overwrite_native_function_implementation");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_overwrite_native_function_implementation");
 }
 
 #[test]
 fn test_u128_i128_as_parameter_and_return_value() {
-	call_wasm_method::<HostFunctions>("test_u128_i128_as_parameter_and_return_value");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_u128_i128_as_parameter_and_return_value");
 }
 
 #[test]
 fn test_vec_return_value_memory_is_freed() {
-	call_wasm_method::<HostFunctions>("test_vec_return_value_memory_is_freed");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_vec_return_value_memory_is_freed");
 }
 
 #[test]
 fn test_encoded_return_value_memory_is_freed() {
-	call_wasm_method::<HostFunctions>("test_encoded_return_value_memory_is_freed");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_encoded_return_value_memory_is_freed");
 }
 
 #[test]
 fn test_array_return_value_memory_is_freed() {
-	call_wasm_method::<HostFunctions>("test_array_return_value_memory_is_freed");
+	call_wasm_method::<HostFunctions>(&WASM_BINARY[..], "test_array_return_value_memory_is_freed");
+}
+
+#[test]
+fn test_verification_of_input() {
+	call_wasm_method::<HostFunctionsDeprecated>(
+		&WASM_BINARY_DEPRECATED[..],
+		"test_verification_of_input_old",
+	);
+	call_wasm_method::<HostFunctions>(
+		&WASM_BINARY[..],
+		"test_verification_of_input_new",
+	);
+}
+
+#[test]
+fn test_verification_of_input_deprecated_wasm_with_newest_host_functions() {
```

Comment:
This can be removed.

Reply:
why? this is the actual test that old wasm is not failing despite we changed implementation

Reply:
The implementations you are testing here are only the same, because you copy pasted code to have 2 different wasm files. It doesn't bring anything to test with the dummy host interface. All the other tests are already testing that the interfaces work as expected.

Reply:
Same answer as https://github.com/paritytech/substrate/pull/5328#discussion_r396354779